source/fuzz/transformation_flatten_conditional_branch.h

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_TRANSFORMATION_FLATTEN_CONDITIONAL_BRANCH_H
+#define SOURCE_FUZZ_TRANSFORMATION_FLATTEN_CONDITIONAL_BRANCH_H
+
+#include "source/fuzz/transformation.h"
+
+namespace spvtools {
+namespace fuzz {
+
+// A structure declaration for all the fresh ids needed to enclose a
+// side-effecting instruction inside a conditional.
+struct IdsForEnclosingInst {
```

Comment:
I suggest you do away with this class altogether and instead do what I've done for AccessChainClampingInfo:

- In the fuzzer pass it's no trouble to build up the protobuf object directly
- In the tests it is a pain to have to do this repeatedly per test, but you can make a factory method - see protobufs::AccessChainClampingInfo MakeAccessClampingInfo(...) in transformation_add_function_test.

I like to avoid classes that mirror the protobuf messages unless there's a really good reason.

Reply:
Done